src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -654,6 +655,43 @@ private void WriteExifChunk(Stream stream, ImageMetadata meta)
             this.WriteChunk(stream, PngChunkType.Exif, meta.ExifProfile.ToByteArray());
         }
 
+        /// <summary>
+        /// Writes an iTXT chunk, containing the XMP metdata to the stream, if such profile is present in the metadata.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="meta">The image metadata.</param>
+        private void WriteXmpChunk(Stream stream, ImageMetadata meta)
+        {
+            if (((this.options.ChunkFilter ?? PngChunkFilter.None) & PngChunkFilter.ExcludeTextChunks) == PngChunkFilter.ExcludeTextChunks)
+            {
+                return;
+            }
+
+            if (meta.XmpProfile is null)
+            {
+                return;
+            }
+
+            meta.XmpProfile.UpdateData();
+            var xmpData = meta.XmpProfile.Data;
+
+            if (xmpData.Length == 0)
+            {
+                return;
+            }
+
+            byte[] payload = new byte[xmpData.Length + PngConstants.XmpKeyword.Length + 5];
+            PngConstants.XmpKeyword.CopyTo(payload);
+            int bytesWritten = PngConstants.XmpKeyword.Length;
+            payload[bytesWritten++] = 0; // Keyword string terminator
```

Comment:
Why zero out here if the array is already zeroed? Can we do this without allocating and use our pool?

Reply:
Indeed, we can just skip over these bytes